mypy/checkexpr.py

```diff
@@ -561,6 +554,21 @@ def visit_call_expr_inner(self, e: CallExpr, allow_none_return: bool = False) ->
             ):
                 member = e.callee.name
                 object_type = self.chk.lookup_type(e.callee.expr)
+
+        if (
+            self.chk.options.disallow_untyped_calls
+            and self.chk.in_checked_function()
+            and isinstance(callee_type, CallableType)
+            and callee_type.implicit
+        ):
+            if fullname is None and member is not None:
+                assert object_type is not None
+                fullname = self.method_fullname(object_type, member)
+            if not fullname or not any(
+                fullname.startswith(p) for p in self.chk.options.untyped_call_exception
```

Comment:
Make sure the exception `re` doesn't apply to `retry`, i.e. the startswith operation should probably be at module name component level.

Reply:
Good catch!